crates/matrix-sdk-ui/src/app/mod.rs

```diff
@@ -0,0 +1,206 @@
+
+                pin_mut!(encryption_sync_stream);
+
+                // Note: any error on one of the two syncs will cause the overall stream to error
+                // and the loop to terminate.
+
+                loop {
+                    tokio::select! {
+                        encryption_sync_result = encryption_sync_stream.next() => {
+                            if let Some(encryption_sync_result) = encryption_sync_result {
+                                if let Err(err) = encryption_sync_result {
+                                    tracing::error!("Encryption sync returned an error: {err:#}");
+                                    state_observer.set(AppState::Error);
+                                    break;
+                                }
+                            } else {
+                                state_observer.set(AppState::Terminated);
+                                break;
+                            }
+                        }
+
+                        room_list_result = room_list_stream.next() => {
+                            if let Some(room_list_result) = room_list_result {
+                                if let Err(err) = room_list_result {
+                                    tracing::error!("Room list sync returned an error: {err:#}");
+                                    state_observer.set(AppState::Error);
+                                    break;
+                                }
+                            } else {
+                                state_observer.set(AppState::Terminated);
+                                break;
+                            }
+                        }
+                    }
+                }
+            } else {
+                while let Some(res) = room_list_stream.next().await {
+                    if let Err(err) = res {
+                        tracing::error!("Error while processing app (room list) state: {err:#}");
+                        state_observer.set(AppState::Error);
+                        break;
+                    }
+                }
+
+                state_observer.set(AppState::Terminated);
+            }
+        }));
+
+        Ok(())
+    }
+
+    pub fn pause(&self) -> Result<(), Error> {
+        self.room_list.stop_sync()?;
+        if let Some(ref encryption_sync) = self.encryption_sync {
+            encryption_sync.stop()?;
+        }
+        Ok(())
+    }
+}
+
+pub struct AppBuilder {
+    client: Client,
+    identifier: String,
+    with_cross_process_lock: bool,
+    with_encryption_sync: bool,
+}
+
+impl AppBuilder {
+    fn new(client: Client) -> Self {
+        Self {
+            client,
+            with_cross_process_lock: false,
+            with_encryption_sync: false,
+            identifier: "app".to_owned(),
+        }
+    }
+
+    pub fn identifier(mut self, name: String) -> Self {
+        self.identifier = name;
+        self
+    }
+
+    pub fn with_encryption_sync(mut self) -> Self {
```

Comment:
As mentioned in the Matrix room, do we need this? Can't we make it always true?

Reply:
I think that's needed until we enable the encryption sync by default... which should be the case any time now.

Reply:
Alright, once we do that, please let us remove this as well.